src/Parser/DeathKnight/Shared/RuneTracker.js

```diff
@@ -60,6 +64,44 @@ class RuneTracker extends Analyzer {
       this.addCharge();
     }
   }
+  on_toPlayer_applybuff(event){
+    if(event.ability.guid === SPELLS.RUNIC_CORRUPTION.id){
+      const multiplier = 1 / (1 + RUNIC_CORRUPTION_INCREASE);
+      RUNE_IDS.forEach((spellId) => {
+        this.changeCooldown(spellId, multiplier);
+      });
+    }
+    if(event.ability.guid === SPELLS.RUNE_MASTER.id){
+      const multiplier = 1 / (1 + T21_4PIECE_BLOOD_INCREASE);
+      RUNE_IDS.forEach((spellId) => {
+        this.changeCooldown(spellId, multiplier);
+      });
+    }
+  }
+  on_toPlayer_removebuff(event){
+    if(event.ability.guid === SPELLS.RUNIC_CORRUPTION.id){
+      const multiplier = 1 + RUNIC_CORRUPTION_INCREASE;
+      RUNE_IDS.forEach((spellId) => {
+        this.changeCooldown(spellId, multiplier);
+      });
+    }
+    if(event.ability.guid === SPELLS.RUNE_MASTER.id){
+      const multiplier = 1 + T21_4PIECE_BLOOD_INCREASE;
+      RUNE_IDS.forEach((spellId) => {
+        this.changeCooldown(spellId, multiplier);
+      });
+    }
+  }
+
+  changeCooldown(spellId, multiplier){
+    if(!this.spellUsable.isOnCooldown(spellId)){
+      return;
+    }
+    const remainingCooldown = this.spellUsable.cooldownRemaining(spellId);
+    const newCooldown = remainingCooldown * multiplier;
+    const reduction = remainingCooldown - newCooldown;
+    this.spellUsable.reduceCooldown(spellId, reduction);
```

Comment:
Won't this throw an error if one of the runes isn't on cooldown?

Reply:
Should be cought in line 97

Reply:
Woooow my eyes slid right over that